### DISCUSSION TITLE
ayuda

### DISCUSSION THREAD

**Discussion by Vx26** [termux/termux-app#3879] [General]
esto aq se debe??  [*] (1) https://mirrors.pku.edu.cn/termux/termux-main/: ok
/data/data/com.termux/files/usr/bin/pkg: line 106: MAIN: unbound variable
/data/data/com.termux/files/usr/bin/pkg: line 111: WEIGHT: unbound variable
/data/data/com.termux/files/usr/bin/pkg: line 106: MAIN: unbound variable
/data/data/com.termux/files/usr/bin/pkg: line 111: WEIGHT: unbound variable

**Comment by agnostic-apollo** [comment]
You have some invalid mirror file under `$PREFIX@/etc/termux/mirrors` which are not exporting the correct variables. Either fix it manually or reinstall `termux-tools` package with `apt install --reinstall termux-tools` and use `Y` option for "install package maintainer's version" if asked.

https://github.com/termux/termux-tools/blob/master/mirrors/default

**Comment by Vx26** [comment]
#aque se debe esto ?   E: Unable to parse package file
/data/data/com.termux/files/usr/var/lib/apt/extended_states (1)

**Reply by agnostic-apollo** [reply]
You can open the file and try to fix issues or just delete it, at the cost of losing auto installed packages info. It should have 2 newline separated entries for packages like following.

```
Package: llvm
Architecture: aarch64
Auto-Installed: 1

Package: zip
Architecture: aarch64
Auto-Installed: 1
```

https://askubuntu.com/questions/229102/what-is-aptitudes-extended-state-information

**Comment by agnostic-apollo** [comment]
```shell
 $ find /data/data/com.termux/files/usr/etc/termux/mirrors -type f -empty
/data/data/com.termux/files/usr/etc/termux/mirrors/china/mirrors.njupt.edu.cn
/data/data/com.termux/files/usr/etc/termux/mirrors/china/mirrors.scau.edu.cn
/data/data/com.termux/files/usr/etc/termux/mirrors/china/mirrors.hit.edu.cn
/data/data/com.termux/files/usr/etc/termux/mirrors/china/mirrors.dgut.edu.cn
```

Seems like https://github.com/termux/termux-tools/commit/138bb91db9009bec78ca82a6d812df597b80dd6c deleted the lines, but not files itself of mirrors removed and didn't delete the entries from `Makefile.am` either.

https://github.com/termux/termux-tools/blob/34fe179068f9e5a4c349e160d539f3737ce242cd/mirrors/Makefile.am#L9

https://github.com/termux/termux-tools/blob/34fe179068f9e5a4c349e160d539f3737ce242cd/mirrors/china/mirrors.njupt.edu.cn

**Reply by agnostic-apollo** [reply]
Should be fixed with [`termux-tools` `v1.41.0`](https://github.com/termux/termux-tools/releases/tag/v1.41.0).
